pkg/kubectl/cmd/patch.go

```diff
@@ -180,6 +180,7 @@ func RunPatch(f cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []strin
 		if !options.Local {
 			dataChangedMsg := "not patched"
 			helper := resource.NewHelper(client, mapping)
+			emptyPatchObj, err := helper.Patch(namespace, name, patchType, []byte("{}"))
```

Comment:
yes, but I've try @liggitt 's pr, it doesn't fix situation like `patch -p '{"no-exist" : "no-exist"} `  or empty patch `patch -p ’{}' `, my hotfix maybe a little clumsy, but it works.